packages/flutter/test/cupertino/radio_test.dart

```diff
@@ -428,4 +429,45 @@ void main() {
     // Release pointer after widget disappeared.
     await gesture.up();
   });
+
+  testWidgets('Radio configures mouse cursor', (WidgetTester tester) async {
+    await tester.pumpWidget(CupertinoApp(
+      home: Center(
+        child: CupertinoRadio<int>(
+          value: 1,
+          groupValue: 1,
+          onChanged: (int? i) { },
+          mouseCursor: SystemMouseCursors.forbidden,
+        ),
+      ),
+    ));
+    final TestGesture gesture = await tester.createGesture(
+      kind: PointerDeviceKind.mouse,
+      pointer: 1
+    );
+    await gesture.addPointer(location: tester.getCenter(find.byType(CupertinoRadio<int>)));
+    await tester.pump();
+    await gesture.moveTo(tester.getCenter(find.byType(CupertinoRadio<int>)));
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1),
+      SystemMouseCursors.forbidden);
+  });
+
+  testWidgets('Radio default mouse cursor', (WidgetTester tester) async {
+    await tester.pumpWidget(CupertinoApp(
+      home: Center(
+        child: CupertinoRadio<int>(
+          value: 1,
+          groupValue: 1,
+          onChanged: (int? i) { },
+        ),
+      ),
+    ));
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse, pointer: 1);
+    await gesture.addPointer(location: tester.getCenter(find.byType(CupertinoRadio<int>)));
+    addTearDown(gesture.removePointer);
+    await tester.pump();
+    await gesture.moveTo(tester.getCenter(find.byType(CupertinoRadio<int>)));
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1),
+        kIsWeb ? SystemMouseCursors.click : SystemMouseCursors.basic);
+  });
```

Comment:
Nit: Maybe also add a test to Radio.adaptive that tests that CupertinoRadio ends up with the mouseCursor.

Reply:
Considering `Radio.adaptive` delegates to `CupertinoRadio` on iOS/macOS (and a test for that exists), combined with the `CupertinoRadio` mouse cursor tests written here, might such a test be necessary?